### PR TITLE
childwindow-proton: Rebase patches and other things

### PR DESCRIPTION
**Pull request by Artewar67**
Should fix this: https://github.com/Frogging-Family/wine-tkg-git/issues/1247

### CODE REVIEW

**Comment by Nanotwerp** [comment]
Tested this PR as a patch and it compiles `Proton-tkg 9.15.r13.gc901884a` successfully!

**Comment by silverhikari** [comment]
after applying the patch to the latest git, i get "/wine-tkg-git/wine-tkg-patches/misc/childwindow/childwindow-proton: line 103: syntax error: unexpected end of file". i applied the patch by going to the base directory that contains proton-tkg and wine-tkg-kit and running "patch -p1 -i 1248.patch "

**Comment by Nanotwerp** [comment]
> after applying the patch to the latest git, i get "/wine-tkg-git/wine-tkg-patches/misc/childwindow/childwindow-proton: line 103: syntax error: unexpected end of file". i applied the patch by going to the base directory that contains proton-tkg and wine-tkg-kit and running "patch -p1 -i 1248.patch "

Try `git apply 1248.patch` instead.

**Comment by silverhikari** [comment]
nope git apply does not change the problem

**Comment by silverhikari** [comment]
just tried checking out the pull the request and still get the error

**Comment by Nanotwerp** [comment]
> just tried checking out the pull request and still get the error

I just tried `git apply 1248.patch` on the base folder, and this is the output:

```
1248.patch:151: trailing whitespace.
@@ -1456,10 +1456,23 @@ 
1248.patch:239: trailing whitespace.
@@ -1456,10 +1456,23 @@ 
1248.patch:329: trailing whitespace.
 
1248.patch:334: trailing whitespace.
 
1248.patch:344: trailing whitespace.
 
warning: squelched 102 whitespace errors
warning: 107 lines add whitespace errors.
```

I think you might have copied and pasted the patch instead of using something like `wget https://github.com/Frogging-Family/wine-tkg-git/pull/1248.patch` (or right-clicking and saving the page of the patch) to download it.

**Comment by silverhikari** [comment]
nope i wget'ed it got the same thing
```1248.patch:151: trailing whitespace.
@@ -1456,10 +1456,23 @@ 
1248.patch:239: trailing whitespace.
@@ -1456,10 +1456,23 @@ 
1248.patch:329: trailing whitespace.
 
1248.patch:334: trailing whitespace.
 
1248.patch:344: trailing whitespace.
 
warning: squelched 102 whitespace errors
warning: 107 lines add whitespace errors.
```

still get the error

**Comment by Nanotwerp** [comment]
> nope i wget'ed it got the same thing
> 
> ```
> 1248.patch:151: trailing whitespace.
> @@ -1456,10 +1456,23 @@ 
> 1248.patch:239: trailing whitespace.
> @@ -1456,10 +1456,23 @@ 
> 1248.patch:329: trailing whitespace.
>  
> 1248.patch:334: trailing whitespace.
>  
> 1248.patch:344: trailing whitespace.
>  
> warning: squelched 102 whitespace errors
> warning: 107 lines add whitespace errors.
> ```
> 
> still get the error

Those are just trailing whitespace warnings. They're not the error that you got before. You should be able to compile now.

**Comment by silverhikari** [comment]
this my complete process https://paste.ee/p/YI90G

File: wine-tkg-git/wine-tkg-patches/misc/childwindow/childwindow-proton [hunk 1]
```diff
@@ -2,8 +2,10 @@
 
 	# Standalone child window support for vk - Fixes World of Final Fantasy and others - https://bugs.winehq.org/show_bug.cgi?id=45277 - legacy patchset for older trees applied at an earlier stage in the script
 	if ( [ "$_childwindow_fix" = "true" ] && [ "$_proton_fs_hack" != "true" ] && [ "$_use_staging" = "true" ] ); then
-	  if git merge-base --is-ancestor c2d46eaa1ef07a2921e5711212e9c5c354f58112 HEAD; then
+	  if git merge-base --is-ancestor cb1c03b926ca01619be02f835f382c1dd7d4a478 HEAD; then
 	    _patchname='childwindow-proton.patch' && _patchmsg="Applied child window for vk patch" && nonuser_patcher
+	  if git merge-base --is-ancestor c2d46eaa1ef07a2921e5711212e9c5c354f58112 HEAD; then
```
Comment (Nanotwerp) [inline wine-tkg-git/wine-tkg-patches/misc/childwindow/childwindow-proton:7]:
```suggestion
          elif git merge-base --is-ancestor cb1c03b926ca01619be02f835f382c1dd7d4a478 HEAD; then
```

File: wine-tkg-git/wine-tkg-patches/misc/childwindow/childwindow-proton [hunk 2]
```diff
@@ -52,8 +54,10 @@
 	fi
 
 	if ( [ "$_childwindow_fix" = "true" ] && [ "$_proton_fs_hack" != "true" ] && [ "$_use_staging" != "true" ] ); then
-	  if git merge-base --is-ancestor c2d46eaa1ef07a2921e5711212e9c5c354f58112 HEAD; then
+	  if git merge-base --is-ancestor cb1c03b926ca01619be02f835f382c1dd7d4a478 HEAD; then
             _patchname='childwindow-proton-mainline.patch' && _patchmsg="Applied child window for vk patch (mainline)" && nonuser_patcher
+	  if git merge-base --is-ancestor c2d46eaa1ef07a2921e5711212e9c5c354f58112 HEAD; then
```
Comment (Nanotwerp) [inline wine-tkg-git/wine-tkg-patches/misc/childwindow/childwindow-proton:59]:
```suggestion
          elif git merge-base --is-ancestor cb1c03b926ca01619be02f835f382c1dd7d4a478 HEAD; then
```

**Review by Nanotwerp — CHANGES_REQUESTED** [review]
These two `if` statements should be turned into `elif` statements. Otherwise, [this error](https://github.com/Frogging-Family/wine-tkg-git/pull/1248#issuecomment-2307981305) occurs.

**Comment by Nanotwerp** [comment]
> this is my complete process https://paste.ee/p/YI90G

Oh, I didn't know you were compiling `Wine TKG` instead of `Proton TKG`! I found the cause of the issue. I gave suggestions that fixes it above. Until @Artewar67 implements the changes, you can apply [this patch](https://gist.github.com/Nanotwerp/addb48b35ff91ef70ff0d0fffe0ef91b).

**Comment by silverhikari** [comment]
yep your fix fixed the issue, thanks!

**Comment by Tk-Glitch** [comment]
Thanks Artewar67 and Nanotwerp ! :frog: :heart:

**Pull request merged**